### ISSUE TITLE
Getting duplicate key error when installing Sentry and using external Postgres replicated database

### ISSUE THREAD

**Issue by Alena86** [getsentry/self-hosted#3352]
### Self-Hosted Version

24.9.0

### CPU Architecture

x86_64

### Docker Version

26.0.2

### Docker Compose Version

2.26.1

### Steps to Reproduce

1. Setup external connection to Postgres database by updating sentry.cong.py: 
`DATABASES = {
    "default": {
        "ENGINE": "sentry.db.postgres",
        "NAME": "sentry",
        "USER": "<username>",
        "PASSWORD": "<password>",
        "HOST": "<postgres_hostname>",   #In our case this would be the load balancer that connects to pgpool before the replicated Postgres with 3 replicas
        "PORT": "5432",
    }
}
`

2. Run the installation: 
`./install.sh --no-report-self-hosted-issues`

### Expected Result

Expecting to have installed version of Sentry that we can work on with no errors.

### Actual Result

Getting the error during the migration-database:
```
Project Uptime Subscriptions: 100% |#                          | ETA:  --:--:--
 OK
  Applying uptime.0008_uptime_url_suffix... OK
  Applying uptime.0009_better_name_for_uptime_rdap_columns... OK
  Applying uptime.0010_remove_uptime_whois_columns_state... OK
  Applying uptime.0011_remove_uptime_whois_columns_db... OK
  Applying workflow_engine.0001_workflow_table... OK
  Applying workflow_engine.0002_data_source... OK
  Applying workflow_engine.0003_detector... OK
  Applying workflow_engine.0004_workflowactions... OK
Traceback (most recent call last):
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 91, in inner
    return func(self, sql, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/base.py", line 84, in execute
    return self.cursor.execute(sql, clean_bad_params(params))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "django_content_type_app_label_model_76bd3d3b_uniq"
DETAIL:  Key (app_label, model)=(auth, permission) already exists.


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 77, in inner
    raise_the_exception(self.db, e)
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 75, in inner
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 18, in inner
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 93, in inner
    raise type(e)(f"{e!r}\nSQL: {sql}").with_traceback(e.__traceback__)
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 91, in inner
    return func(self, sql, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/base.py", line 84, in execute
    return self.cursor.execute(sql, clean_bad_params(params))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

psycopg2.errors.UniqueViolation: UniqueViolation('duplicate key value violates unique constraint "django_content_type_app_label_model_76bd3d3b_uniq"\nDETAIL:  Key (app_label, model)=(auth, permission) already exists.\n')
SQL: INSERT INTO "django_content_type" ("app_label", "model") VALUES (%s, %s) RETURNING "django_content_type"."id"

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/.venv/bin/sentry", line 4, in <module>
    raise SystemExit(main())
                     ^^^^^^
  File "/usr/src/sentry/src/sentry/runner/main.py", line 149, in main
    func(**kwargs)
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/runner/decorators.py", line 35, in inner
    return ctx.invoke(f, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/runner/commands/upgrade.py", line 147, in upgrade
    _upgrade(
  File "/usr/src/sentry/src/sentry/runner/commands/upgrade.py", line 63, in _upgrade
    dj_call_command(
  File "/.venv/lib/python3.12/site-packages/django/core/management/__init__.py", line 194, in call_command
    return command.execute(*args, **defaults)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/core/management/base.py", line 459, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/core/management/base.py", line 107, in wrapper
    res = handle_func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/core/management/commands/migrate.py", line 384, in handle
    emit_post_migrate_signal(
  File "/.venv/lib/python3.12/site-packages/django/core/management/sql.py", line 52, in emit_post_migrate_signal
    models.signals.post_migrate.send(
  File "/.venv/lib/python3.12/site-packages/django/dispatch/dispatcher.py", line 189, in send
    response = receiver(signal=self, sender=sender, **named)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/contrib/auth/management/__init__.py", line 78, in create_permissions
    ctypes = ContentType.objects.db_manager(using).get_for_models(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/contrib/contenttypes/models.py", line 98, in get_for_models
    ct = self.create(app_label=app_label, model=model_name)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/models/query.py", line 679, in create
    obj.save(force_insert=True, using=self.db)
  File "/.venv/lib/python3.12/site-packages/django/db/models/base.py", line 891, in save
    self.save_base(
  File "/.venv/lib/python3.12/site-packages/django/db/models/base.py", line 997, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/models/base.py", line 1160, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/models/base.py", line 1201, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/models/query.py", line 1847, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1836, in execute_sql
    cursor.execute(sql, params)
  File "/.venv/lib/python3.12/site-packages/sentry_sdk/utils.py", line 1720, in runner
    return sentry_patched_function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/django/__init__.py", line 651, in execute
    result = real_execute(self, sql, params)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "/.venv/lib/python3.12/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 77, in inner
    raise_the_exception(self.db, e)
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 75, in inner
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 18, in inner
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 93, in inner
    raise type(e)(f"{e!r}\nSQL: {sql}").with_traceback(e.__traceback__)
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 91, in inner
    return func(self, sql, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/base.py", line 84, in execute
    return self.cursor.execute(sql, clean_bad_params(params))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.IntegrityError: UniqueViolation('duplicate key value violates unique constraint "django_content_type_app_label_model_76bd3d3b_uniq"\nDETAIL:  Key (app_label, model)=(auth, permission) already exists.\n')
SQL: INSERT INTO "django_content_type" ("app_label", "model") VALUES (%s, %s) RETURNING "django_content_type"."id"
```

**Comment by bijancot** [comment]
Actually this issue happened sometimes in the past. My only work around is deleting that constraint for a moment and if possible i will recover the constraint. Because there is one table that is user table that has duplicate email constraint that in the end i'm not recovering that one

**Comment by Alena86** [comment]
I'm sorry but I'm not sure what are you suggesting... 
This is happening during the installation... so I cannot really stop it in the middle of the installation.. unless I'm running each sh script individually? 

I managed to remove completely the postgres container and connect straight to our external postgres bu now I'm getting this error: 
```
Saved Searchs: 100% |#                                         | ETA:  --:--:--
 OK
  Applying sentry.0503_alter_notification_actor_nullable... OK
  Applying sentry.0504_add_artifact_bundle_index... OK
  Applying sentry.0505_debugfile_date_accessed... OK
  Applying sentry.0506_null_boolean_fields... OK
Rules: N/A% |                                                  | ETA:  --:--:--Traceback (most recent call last):
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 91, in inner
    return func(self, sql, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/base.py", line 84, in execute
    return self.cursor.execute(sql, clean_bad_params(params))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.UndefinedColumn: column sentry_rule.owner_id does not exist
LINE 1: ...y_rule"."date_added", "sentry_rule"."project_id", "sentry_ru...
                                                             ^


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 77, in inner
    raise_the_exception(self.db, e)
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 75, in inner
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 18, in inner
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 93, in inner
    raise type(e)(f"{e!r}\nSQL: {sql}").with_traceback(e.__traceback__)
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 91, in inner
    return func(self, sql, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/base.py", line 84, in execute
    return self.cursor.execute(sql, clean_bad_params(params))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.UndefinedColumn: UndefinedColumn('column sentry_rule.owner_id does not exist\nLINE 1: ...y_rule"."date_added", "sentry_rule"."project_id", "sentry_ru...\n                                                             ^\n')
SQL: SELECT "sentry_rule"."id", "sentry_rule"."environment_id", "sentry_rule"."label", "sentry_rule"."data", "sentry_rule"."status", "sentry_rule"."date_added", "sentry_rule"."project_id", "sentry_rule"."owner_id", "sentry_rule"."source" FROM "sentry_rule" ORDER BY "sentry_rule"."id" ASC LIMIT 1000

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/.venv/bin/sentry", line 4, in <module>
    raise SystemExit(main())
                     ^^^^^^
  File "/usr/src/sentry/src/sentry/runner/main.py", line 149, in main
    func(**kwargs)
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/runner/decorators.py", line 35, in inner
    return ctx.invoke(f, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/runner/commands/upgrade.py", line 147, in upgrade
    _upgrade(
  File "/usr/src/sentry/src/sentry/runner/commands/upgrade.py", line 63, in _upgrade
    dj_call_command(
  File "/.venv/lib/python3.12/site-packages/django/core/management/__init__.py", line 194, in call_command
    return command.execute(*args, **defaults)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/core/management/base.py", line 459, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/core/management/base.py", line 107, in wrapper
    res = handle_func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/core/management/commands/migrate.py", line 357, in handle
    post_migrate_state = executor.migrate(
                         ^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/migrations/executor.py", line 135, in migrate
    state = self._migrate_all_forwards(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/migrations/executor.py", line 167, in _migrate_all_forwards
    state = self.apply_migration(
            ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/new_migrations/monkey/executor.py", line 141, in apply_migration
    return super().apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/migrations/executor.py", line 255, in apply_migration
    state = migration.apply(state, schema_editor)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/new_migrations/migrations.py", line 22, in apply
    return super().apply(project_state, schema_editor, collect_sql)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/migrations/migration.py", line 132, in apply
    operation.database_forwards(
  File "/.venv/lib/python3.12/site-packages/django/db/migrations/operations/special.py", line 196, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/usr/src/sentry/src/sentry/migrations/0507_delete_pending_deletion_rules.py", line 50, in delete_rules
    for rule in RangeQuerySetWrapperWithProgressBar(Rule.objects.all()):
  File "/usr/src/sentry/src/sentry/utils/query.py", line 244, in __iter__
    for idx, item in enumerate(self.iterator):
  File "/usr/src/sentry/src/sentry/utils/query.py", line 159, in __iter__
    results = list(results[0 : self.step])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/models/query.py", line 400, in __iter__
    self._fetch_all()
  File "/.venv/lib/python3.12/site-packages/django/db/models/query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/models/query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
  File "/.venv/lib/python3.12/site-packages/sentry_sdk/utils.py", line 1720, in runner
    return sentry_patched_function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/django/__init__.py", line 651, in execute
    result = real_execute(self, sql, params)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "/.venv/lib/python3.12/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/.venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 77, in inner
    raise_the_exception(self.db, e)
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 75, in inner
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 18, in inner
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 93, in inner
    raise type(e)(f"{e!r}\nSQL: {sql}").with_traceback(e.__traceback__)
  File "/usr/src/sentry/src/sentry/db/postgres/decorators.py", line 91, in inner
    return func(self, sql, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/db/postgres/base.py", line 84, in execute
    return self.cursor.execute(sql, clean_bad_params(params))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.ProgrammingError: UndefinedColumn('column sentry_rule.owner_id does not exist\nLINE 1: ...y_rule"."date_added", "sentry_rule"."project_id", "sentry_ru...\n                                                             ^\n')
SQL: SELECT "sentry_rule"."id", "sentry_rule"."environment_id", "sentry_rule"."label", "sentry_rule"."data", "sentry_rule"."status", "sentry_rule"."date_added", "sentry_rule"."project_id", "sentry_rule"."owner_id", "sentry_rule"."source" FROM "sentry_rule" ORDER BY "sentry_rule"."id" ASC LIMIT 1000
Error in install/set-up-and-migrate-database.sh:21.
'$dcr web upgrade --create-kafka-topics' exited with status 1
-> ./install.sh:main:37
--> install/set-up-and-migrate-database.sh:source:21

Cleaning up...
```

**Comment by bijancot** [comment]
yep sorry, i thought you trying to restore your sentry to 24.9.0 because that kind of error happen when do a backup restore.  Your first error is kinda happen because you are trying a fresh install inside a database that already have sentry data in it(?) and the seconds seems the data is nowhere to be found. Maybe trying to check .env will help to make sure which version you use etc. 

Or maybe share your config regarding external database will also help

**Comment by Alena86** [comment]
Ooooh no... I wish LOL not at that state yet.
1. First issue: 
It is a clean machine with docker and git in it, where I'm trying to install the Sentry with docker-compose. 
Since we already have Postgres/Kafka/Redis running, I wanted to use that as external as I already have them running in our systems and that's where I'm getting that error. 
```
django.db.utils.IntegrityError: UniqueViolation('duplicate key value violates unique constraint "django_content_type_app_label_model_76bd3d3b_uniq"\nDETAIL:  Key (app_label, model)=(auth, permission) already exists.\n')
SQL: INSERT INTO "django_content_type" ("app_label", "model") VALUES (%s, %s) RETURNING "django_content_type"."id"
```
We are running postgres 16 in replicated state with pgpool on top of it. 
And the place where I updated the use of external database is in sentry/sentry.config.py

DATABASES = {
    "default": {
        "ENGINE": "sentry.db.postgres",
        "NAME": "sentry",
        "USER": "<admin_username>",
        "PASSWORD": "<admin_password>",
        "HOST": "postgres.ourdomain.com",
        "PORT": "5432",
    }
}

postgres.ourdomain.com is pointing to the pgpool connecting to the replicated Postgres. 


2. The second issue:
I removed the Postgres container use fully, and redirecting snuba to use existing external Postgres to see if that will fix the previous error.
So I'm guessing in this, the setup of snuba doesn't create those tables properly?? 

Fun fact, when I get into the database, I can see that the column exists in the table just no data in it. 

If you have any idea of how to fix 1st issue or any more info that we might want to know that would be good and very much appreciated :)

**Comment by aldy505** [comment]
Have you check over this issue comment to check if it matches your problem? https://github.com/getsentry/self-hosted/issues/3167#issuecomment-2223701471

> I removed the Postgres container use fully, and redirecting snuba to use existing external Postgres to see if that will fix the previous error.
> So I'm guessing in this, the setup of snuba doesn't create those tables properly??

No, snuba only connects to ClickHouse, only containers that uses the sentry image connects to Postgres.

The command that runs the migration (Postgres and Kafka) is here: https://github.com/getsentry/self-hosted/blob/8887a82a86153d0c9696463c0525976cf8e3f136/install/set-up-and-migrate-database.sh#L36

**Comment by Alena86** [comment]
Hey @aldy505 thank you for your answer, suggestion and clarification. 
We left the sh script to run the reindexing of the databaseas well as changing the lag settings on the Postgres database itself and that did the trick! Thank you!
Curiousity... 
If snuba connects only to the clickhouse, then why in docker-compose, it is waiting on the healthiness of other dbs? 

In particular this portion of the docker-compose? 
```
x-snuba-defaults: &snuba_defaults
  <<: *restart_policy
  depends_on:
    clickhouse:
      <<: *depends_on-healthy
    kafka:
      <<: *depends_on-healthy
    redis:
      <<: *depends_on-healthy
 
```
As now I'm at the state that I have removed redis and postgres containers, but still have Kafka one that is bothering me from snuba that I was gonna open a new issue for? 

Thank you for your response and help! Really appreciate it.

**Comment by Alena86** [comment]
And I guess we can close this issue as we resolved it!

Thank you once again to the community for the help in solving this issue :)

**Comment by aldy505** [comment]
> If snuba connects only to the clickhouse, then why in docker-compose, it is waiting on the healthiness of other dbs? 

I never count Kafka and Redis as a database though, since they are not designed for persistence use cases. Two databases that I'm talking about is Postgres and Clickhouse. 

> As now I'm at the state that I have removed redis and postgres containers, but still have Kafka one that is bothering me from snuba that I was gonna open a new issue for? 

Yes feel free!
